toolchain/check/cpp/thunk.cpp

```diff
@@ -669,9 +669,25 @@ auto PerformCppThunkCall(Context& context, SemIR::LocId loc_id,
                           context.insts().Get(return_slot_id).type_id())),
          .lvalue_id = return_slot_id});
     thunk_arg_ids.push_back(arg_id);
+  } else if (return_slot_id.has_value()) {
+    thunk_arg_ids.push_back(return_slot_id);
   }
 
-  auto result_id = PerformCall(context, loc_id, thunk_callee_id, thunk_arg_ids);
+  // Compute the return type of the call to the thunk.
+  auto thunk_return_type_id =
+      thunk_function.GetDeclaredReturnType(context.sem_ir());
+  if (!thunk_return_type_id.has_value()) {
+    thunk_return_type_id = GetTupleType(context, {});
+    CARBON_CHECK(thunk_takes_return_address || !return_type_id.has_value());
```

Comment:
```suggestion
    CARBON_CHECK(thunk_takes_return_address || !return_type_id.has_value());
    thunk_return_type_id = GetTupleType(context, {});
```
Can the `CHECK` be moved above the assignment? (establish the invariant as early as possible - and also I misread this code a few times missing the difference between `thunk_return_type_id` and `return_type_id` & was confused why `return_type_id` would ever not have a value when it (really `think_return_type_id`) had been assigned to on the previous line)